ngraph/test/backend/unary_test.hpp

```diff
@@ -0,0 +1,211 @@
+        if (tol.using_bits)
+        {
+            if (tol.tolerance_bits >= 0)
+                test_case.run(tol.tolerance_bits);
+            else
+                test_case.run();
+        }
+        else
+        {
+            test_case.run_with_tolerance_as_fp(tol.tolerance_fp);
+        }
+    }
+
+    // overloading support passing reference function instead of expected results
+    template <typename TestEngine,
+              element::Type_t element_type,
+              typename value_type = fundamental_type_for<element_type>>
+    void test_unary(std::shared_ptr<Function> f,
+                    const std::vector<value_type>& input,
+                    value_type (*func)(value_type),
+                    ArgShape ashape = {},
+                    ArgTolerance tol = {})
+    {
+        std::vector<value_type> expected;
+        for (value_type x : input)
+            expected.push_back(func(x));
+
+        test_unary<TestEngine, element_type>(f, input, expected, ashape, tol);
+    }
+
+    // overloading support passing creator instead of ngraph function
+    // Creator is any callable of following prototype:
+    //
+    //          std::shared_ptr<Function> (const ngraph::element::Type&, const PartialShape&)
+    //
+    // unary_func() is a helper function to generate a lambda as callable on simple unary ops with
+    // optional arguments captured in the that lambda.
+    //
+    using UnaryFuncCreator =
+        std::function<std::shared_ptr<Function>(const ngraph::element::Type&, const PartialShape&)>;
+
+    template <typename OpType, typename... Args>
+    typename std::enable_if<std::is_base_of<op::Op, OpType>::value, UnaryFuncCreator>::type
+        unary_func(Args&&... args)
+    {
+        return [args...](const ngraph::element::Type& ele_type, const PartialShape& pshape) {
+            auto X = make_shared<op::Parameter>(ele_type, pshape);
+            auto Y = make_shared<OpType>(X, args...);
+            return make_shared<Function>(Y, ParameterVector{X});
+        };
+    }
+
+    template <typename TestEngine,
+              element::Type_t element_type,
+              typename Creator,
+              typename value_type = fundamental_type_for<element_type>>
+    void test_unary(Creator creator,
+                    const std::vector<value_type>& input,
+                    const std::vector<value_type>& expected,
+                    ArgShape ashape = {},
+                    ArgTolerance tol = {})
+    {
+        if (!ashape.initialzed)
+        {
+            ashape = ArgShape(Shape({input.size()}));
+        }
+        test_unary<TestEngine, element_type>(
+            creator(element_type, ashape.dynamic_shape), input, expected, ashape, tol);
+    }
+
+    template <typename TestEngine,
+              element::Type_t element_type,
+              typename Creator,
+              typename value_type = fundamental_type_for<element_type>>
+    void test_unary(Creator creator,
```

Comment:
IMHO we should add function which return `expected` values from `inputs` and use only one `test_unary`

Reply:
I'd keep this overload based on the same reason above